src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    public class DirectoryInfo_CreateSubDirectory : FileSystemTest
+    {
+        #region UniversalTests
+
+        [Fact]
+        public void NullAsPath_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() => new DirectoryInfo(TestDirectory).CreateSubdirectory(null));
+        }
+
+        [Fact]
+        public void EmptyAsPath_ThrowsArgumentException()
+        {
+            Assert.Throws<ArgumentException>(() => new DirectoryInfo(TestDirectory).CreateSubdirectory(string.Empty));
+        }
+
+        [Fact]
+        public void PathAlreadyExistsAsFile()
+        {
+            string path = GetTestFileName();
+            File.Create(Path.Combine(TestDirectory, path)).Dispose();
+
+            Assert.Throws<IOException>(() => new DirectoryInfo(TestDirectory).CreateSubdirectory(path));
+            Assert.Throws<IOException>(() => new DirectoryInfo(TestDirectory).CreateSubdirectory(IOServices.AddTrailingSlashIfNeeded(path)));
+            Assert.Throws<IOException>(() => new DirectoryInfo(TestDirectory).CreateSubdirectory(IOServices.RemoveTrailingSlash(path)));
+        }
+
+        [Theory]
+        [InlineData(FileAttributes.Hidden)]
+        [InlineData(FileAttributes.ReadOnly)]
+        [InlineData(FileAttributes.Normal)]
+        public void PathAlreadyExistsAsDirectory(FileAttributes attributes)
+        {
+            string path = GetTestFileName();
+            DirectoryInfo testDir = Directory.CreateDirectory(Path.Combine(TestDirectory, path));
+            testDir.Attributes = attributes;
```

Comment:
Looks like a bunch of these tests are basically the same as the ones I left a few comments on earlier, so I won't duplicate those here.